decimal64.go

```diff
@@ -70,6 +75,91 @@ var powersOf10 = []uint64{
 	10000000000000000000,
 }
 
+func (dec *decParts) round(rnd roundContext) uint64 {
+	if rnd.rndStatus == UNSET {
+		return dec.significand
+	}
+	switch rnd.roundType {
+	case roundHalfUp:
+		if rnd.rndStatus == LT || rnd.rndStatus == ZR {
+			return dec.significand
+		}
+		return dec.significand + 1
+	case roundDown:
+		return dec.significand
+	case roundFloor:
+		return dec.significand
+	case roundCeiling: //TODO: fine tune ceiling, floor rounding
+		if rnd.rndStatus == ZR {
+			return dec.significand
+		}
+		if rnd.rndStatus != ZR && dec.sign == 0 {
```

Comment:
The second and third `if` statements should be in an else of the first one.

Reply:
Fixed but also commented out roundCeiling; not working correctly yet